packages/framework/src/Component/Image/ImageFacade.php

```diff
@@ -529,4 +542,32 @@ public function getImagesByEntityId(int $id, string $entityClass): array
 
         return $this->getImagesByEntityIdAndNameIndexedById($id, $entityName, null);
     }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Component\FileUpload\ImageUploadData $imageUploadData
+     * @param object $entity
+     * @param string|null $type
+     */
+    public function setImageUploadDataForEntityAndType(ImageUploadData $imageUploadData, object $entity, ?string $type = null): void
```

Comment:
I was thining before why we don't have an ImageUploadDataFactory to build ImageUploadData and instead it builds sequentially in three places (eg: TransportData, TransportDataFactory, ImageFacade). Could we dedicate the whole thing to the factory?

Reply:
Well I have reworked it as you wished